pkg/controllers/work/apply_controller.go

```diff
@@ -383,13 +389,46 @@ func (r *ApplyWorkReconciler) applyUnstructured(ctx context.Context, gvr schema.
 	}
 
 	// We only try to update the object if its spec hash value has changed.
-	if manifestObj.GetAnnotations()[manifestHashAnnotation] != curObj.GetAnnotations()[manifestHashAnnotation] {
+	if manifestObj.GetAnnotations()[ManifestHashAnnotation] != curObj.GetAnnotations()[ManifestHashAnnotation] {
+		// we need to merge the owner reference between the current and the manifest since we support one manifest
+		// belong to multiple work so it contains the union of all the appliedWork.
+		manifestObj.SetOwnerReferences(mergeOwnerReference(curObj.GetOwnerReferences(), manifestObj.GetOwnerReferences()))
+		// record the raw manifest with the hash annotation in the manifest.
+		if err := setModifiedConfigurationAnnotation(manifestObj); err != nil {
+			return nil, ManifestNoChangeAction, err
+		}
+		annotations := manifestObj.GetAnnotations()
+		if annotations[LastAppliedConfigAnnotation] == "" {
+			klog.V(2).InfoS("using server side apply for manifest", "gvr", gvr, "manifest", manifestRef)
+			return r.applyObject(ctx, gvr, manifestObj)
+		}
+		klog.V(2).InfoS("using three way merge for manifest", "gvr", gvr, "manifest", manifestRef)
 		return r.patchCurrentResource(ctx, gvr, manifestObj, curObj)
 	}
 
 	return curObj, ManifestNoChangeAction, nil
 }
 
+// applyObject uses server side apply to apply the manifest.
+func (r *ApplyWorkReconciler) applyObject(ctx context.Context, gvr schema.GroupVersionResource,
+	manifestObj *unstructured.Unstructured) (*unstructured.Unstructured, applyAction, error) {
+	manifestRef := klog.ObjectRef{
+		Name:      manifestObj.GetName(),
+		Namespace: manifestObj.GetNamespace(),
+	}
+	options := metav1.ApplyOptions{
+		FieldManager: workFieldManagerName,
+		Force:        true,
+	}
+	manifestObj, err := r.spokeDynamicClient.Resource(gvr).Namespace(manifestObj.GetNamespace()).Apply(ctx, manifestObj.GetName(), manifestObj, options)
```

Comment:
I'm using server side apply because,

- It forces conflicts on fields which is managed by the controller similar to three way merge patch
- It doesn't care about fields changed by users on the member cluster side